### PR TITLE
fix(schematics): migrate spec to skipTest to be in line with Angular CLI

### LINKED ISSUE #2242: Schematics: use `skipTests` instead of `spec`
With the new version of the Angular CLI, the `spec` option from the schematics has been removed. Instead, a `skipTests` option has been added.

We would like to keep our schematics in line with the Angular CLI schematics.

This change includes:

- [ ] Modifying the schematic options
- [ ] Update the schematics code
- [ ] Uppdate the docs

For ref: angular/angular-cli@e92c46a

### PR DESCRIPTION
**Pull request by Jefiozie**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
You need to use the `spec` option for generating test files. 

Closes #2242 

## What is the new behavior?
We are now inline with the Angular CLI `skipTest` option.
## Does this PR introduce a breaking change?

```
[X] Yes
[ ] No
```

### CODE REVIEW

**Comment by Jefiozie** [comment]
cc: @timdeschryver

**Comment by Jefiozie** [comment]
Thanks for looking into it I will update the PR soon with your findings.

**Comment by Jefiozie** [comment]
So I changed as I believe the correct things now and made updates to the docs. Hope that these are the correct changes.

**Comment by Jefiozie** [comment]
Okay, I don't exactly know why the bazel tests are failing. Is it because of my changes?

File: modules/schematics/src/feature/index.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export default function(options: FeatureOptions): Rule {
         name: options.name,
         path: options.path,
         project: options.project,
-        spec: false,
+        skipTest: true,
```
Comment (timdeschryver) [inline modules/schematics/src/feature/index.ts]: Not for now, but could we introduce a `skipTest` flag for the `feature` schematic @brandonroberts ?
By default, we create spec files for every schematic, but not for this schematic.

**Comment by timdeschryver** [comment]
@Jefiozie I don't think it's because of the changes, let me re-trigger the CI

**Comment by Jefiozie** [comment]
Reverted the `docs` changes, I think we are good now for this PR 👍

**Comment by timdeschryver** [comment]
@Jefiozie could you also add a breaking change note to the PR please, you can find a template in [CONTRIBUTING.MD](https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#footer)

**Comment by Jefiozie** [comment]
I added the breaking changes request.

**Comment by brandonroberts** [comment]
@Jefiozie will you resolve the merge conflicts? Thanks!

**Comment by Jefiozie** [comment]
I did the merge on my tablet I see that there are problems. Will fix this later this week.

**Comment by Jefiozie** [comment]
@brandonroberts, I merged all the latest changes into my branch. At first I thought the failing tests where because of my changes however I believe that it is because of recent changes in the master branch around #2301. 

Could you o maybe @alex-okrushko verify this? (would love to know this for sure 😄 )

**Comment by timdeschryver** [comment]
I also noticed the failing build, should be resolved if https://github.com/ngrx/platform/pull/2313 is merged.

**Comment by Jefiozie** [comment]
Thanks @timdeschryver

**Comment by Jefiozie** [comment]
@brandonroberts i think it's good to merge now.

**Comment by Jefiozie** [comment]
@timdeschryver, I believe it should be correct now.

**Pull request merged**